cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -1139,25 +1139,80 @@ int64_t get_batch_nbytes(const arrow::RecordBatch& rb) {
   return accumulated;
 }
 
+velox::RowVectorPtr RecordBatch2VeloxRowVector(const arrow::RecordBatch& rb) {
+  ArrowArray arrowArray;
+  ArrowSchema arrowSchema;
+  GLUTEN_THROW_NOT_OK(arrow::ExportRecordBatch(rb, &arrowArray, &arrowSchema));
+  auto vp =
+      velox::importFromArrowAsOwner(arrowSchema, arrowArray, gluten::GetDefaultLeafWrappedVeloxMemoryPool().get());
+  return std::dynamic_pointer_cast<velox::RowVector>(vp);
+}
+
 } // namespace
 
+std::shared_ptr<arrow::Buffer> VeloxShuffleWriter::serialize(
+    const RowVectorPtr& rowVector,
+    const VectorSerde::Options* serdeOptions) {
+  auto numRows = rowVector->size();
+
+  std::vector<IndexRange> rows(numRows);
+  for (int i = 0; i < numRows; i++) {
+    rows[i] = IndexRange{i, 1};
+  }
+
+  std::shared_ptr<arrow::ResizableBuffer> value_buffer;
+  GLUTEN_ASSIGN_OR_THROW(
+      value_buffer, arrow::AllocateResizableBuffer(rowVector->estimateFlatSize() * 1.1, options_.memory_pool.get()));
+  auto output = std::make_shared<arrow::io::BufferOutputStream>(value_buffer);
+
+  auto arena = std::make_unique<StreamArena>(veloxPool_);
+  auto rowType = asRowType(rowVector->type());
+  auto serializer = serde_->createSerializer(rowType, numRows, arena.get(), serdeOptions);
```

Comment:
Question: Any consideration of not using ```UnsafeRowVectorSerializer```? It also can append new row vector? 

Did we compare the micro-benchmark e.g., ```ShuffleSplitBenchmark``` to compare ```ArrowShuffleWriter``` and ```VeloxShuffleWriter```? Also any space efficiency when adopting different serialization?

Reply:
Both of these 2 serializer will copy the data one by one to several big buffers, and then copy the big buffers to OutputStream.
I will implement new SparkSerializer.
Meituan will help to split velox to velox, and new SparkSerializer will serialize the velox RowVector buffer, and write to one big buffer with only once memory copy.